lib/web_ui/lib/src/engine/window.dart

```diff
@@ -202,6 +197,69 @@ base class EngineFlutterView implements ui.FlutterView {
   final DimensionsProvider dimensionsProvider;
 
   Stream<ui.Size?> get onResize => dimensionsProvider.onResize;
+
+  /// Called immediately after the view has been resized.
+  ///
+  /// When there is a text editing going on in mobile devices, do not change
+  /// the physicalSize, change the [window.viewInsets]. See:
+  /// https://api.flutter.dev/flutter/dart-ui/FlutterView/viewInsets.html
+  /// https://api.flutter.dev/flutter/dart-ui/FlutterView/physicalSize.html
+  ///
+  /// Note: always check for rotations for a mobile device. Update the physical
+  /// size if the change is caused by a rotation.
+  void _didResize(ui.Size? newSize) {
+    StyleManager.scaleSemanticsHost(dom.semanticsHost, devicePixelRatio);
+    final ui.Size newPhysicalSize = _computePhysicalSize();
+    final bool isEditingOnMobile =
+        isMobile && !_isRotation(newPhysicalSize) && textEditing.isEditing;
+    if (isEditingOnMobile) {
+      _computeOnScreenKeyboardInsets(true);
+    } else {
+      _physicalSize = newPhysicalSize;
+      // When physical size changes this value has to be recalculated.
+      _computeOnScreenKeyboardInsets(false);
+    }
```

Comment:
I'm 99% sure this bit only matters when the app is embedded in full-screen mode (the only one that really computes keyboard insets).

Can we make this `didResize` something that is overridable in the Implicit view + full-screen implementation, and move this version of the logic there?

The case of embedded views in custom elements shouldn't worry about this. WDYT?

Reply:
What you are suggesting makes sense. But I would like to do it in a separate PR when I have the time to refactor it and test it properly.